pixeldata/src/gdcm.rs

```diff
@@ -55,7 +56,26 @@ where
         let rescale_slope = rescale_slope(self);
         let number_of_frames = number_of_frames(self).context(GetAttributeSnafu)?;
         let voi_lut_function = voi_lut_function(self).context(GetAttributeSnafu)?;
-        let voi_lut_function = voi_lut_function.and_then(|v| VoiLutFunction::try_from(&*v).ok());
+        let voi_lut_function: Option<Vec<VoiLutFunction>> = voi_lut_function.
+            and_then(|fns| fns.iter()
+                .map(|v| VoiLutFunction::try_from((*v).as_str()).ok())
+                .collect()
+            );
+        if let Some(inner) = &voi_lut_function {
+            if !(inner.len() == number_of_frames as usize || inner.len() == 1) {
+                LengthMismatchVoiLutFunctionSnafu {
+                    vm: inner.len() as u32,
+                    nr_frames: number_of_frames as u32
+                }.fail().context(GetAttributeSnafu)?;
+            } 
```

Comment:
I'm not super happy with this, I wanted to use the `ensure` macro like I did in `lib.rs`, but I couldn't figure out a way to get it to work...

Reply:
`ImagingProperties` in `pixeldata/src/lib.rs` returns a `Result<Self, attribute::GetAttributeError>`, if I moved the `LengthMismatch*` errors up to `InnerError` I would need to change change the return signature of `ImagingProperties` and then add the `fail().context(GetAttributeSnafu)` to all the fallible operations in `from_obj`, which seems to me more code changes.  

I'm happy to go with either, just lmk which interface you would prefer, or if I'm missing something!